### DISCUSSION TITLE
Running a signed transaction is not working  is not working when I run `delegateBySig` transaction

### DISCUSSION THREAD

**Discussion by ntourne** [wevm/wagmi#2167] [Q&A] [answered]
I'm working on a feature to allow accounts to delegate tokens in a DAO without paying gas. So, then a relayer will execute that proposal on-chain.

The user needs to sign a message (EIP-712) like this:

```typescript
const {
    data: dataSignature,
    isError,
    isLoading,
    isSuccess,
    signTypedDataAsync,
  } = useSignTypedData()

...

const nonce = 0

const message = {
  delegate: "...",
  nonce: 0,
  expiry: expiry.toString(),
}

const domain = {
  name: "My App",
  version: "1.0",
  chainId: chainId,
  verifyingContract: governorAddress as `0x${string}`,
}

const types = {
  EIP712Domain: [
    { name: "name", type: "string" },
    { name: "version", type: "string" },
    { name: "chainId", type: "uint256" },
    { name: "verifyingContract", type: "address" },
  ],
  Delegation: [
    { name: "delegate", type: "address" },
    { name: "nonce", type: "uint256" },
    { name: "expiry", type: "uint256" },
  ],
}

const signature = await signTypedDataAsync({
  domain,
  types,
  value: message,
})

const { v, r, s } = ethers.utils.splitSignature(signature)
```

Then, once the signature I send on chain using (where `send` is a helper to just send transactions on chain).
```typescript
send(() => erc20.delegateBySig(delegateeAddress, nonce, expiry, v, r, s))
```

The transaction seems to be executed successfully but there are two issues:
-  The `DelegateChanged` event doesn't work as expected. The `fromDelegate` value is `0` and it should be the address of the signer (see screenshot).
- There is no update in the state of the contract.
<img width="1189" alt="image" src="https://user-images.githubusercontent.com/3385523/231339449-df009416-68af-4e2d-ac02-0dbbec7caab4.png">

Any clue what could be wrong? Thanks!

**Comment by jxom** [comment]
Is `chainId` meant to be a string? You are defining it as a `uint256` in your schema, but it is a string in the domain. 

If converting to a number doesn't work – it could be an ethers.js related issue (we don't prepare typed data payloads or explicitly parse signatures in our implementation, we just forward it to ethers.js).

**Reply by ntourne** [reply]
> Is `chainId` meant to be a string? You are defining it as a `uint256` in your schema, but it is a string in the domain.
> 
> If converting to a number doesn't work – it could be an ethers.js related issue (we don't prepare typed data payloads or explicitly parse signatures in our implementation, we just forward it to ethers.js).

Thank, @jxom. I tried `chainId` as number and string but I got the same result. Going to check out ethers.js related issues.

**Comment by ntourne** [comment, marked as answer]
I found the issue. It was not related to wagmi or ethers, but to the `name` I used on the `domain` separator.
The correct value is the ERC20 contract's `name` and not a hardcoded value (for example: `"My App"`).

```typescript
const domain = {
  name: "<ERC20 contract name>" ,  // <--- this fixes the issue
  version: "1.0",
  chainId: chainId,
  verifyingContract: governorAddress as `0x${string}`,
}
```
